src/test/metrics_api_test.cc

```diff
@@ -232,6 +233,64 @@ MetricAPIHelper(TRITONSERVER_Metric* metric, TRITONSERVER_MetricKind kind)
   TRITONSERVER_ErrorDelete(err);
 }
 
+void
+HistogramAPIHelper(
+    TRITONSERVER_Server* server, TRITONSERVER_Metric* metric,
+    std::vector<double> buckets, std::string metric_name,
+    std::string labels_str)
+{
+  // Observe data
+  std::vector<double> data{0.05, 1.5, 6.0};
+  double sum = 0.0;
+  for (auto datum : data) {
+    FAIL_TEST_IF_ERR(
+        TRITONSERVER_MetricObserve(metric, datum), "observe metric value");
+    sum += datum;
+  }
+  std::vector<std::uint64_t> cumulative_counts = {1, 1, 2, 2, 3, 3};
+  ASSERT_EQ(buckets.size() + 1, cumulative_counts.size());
```

Comment:
The `cumulative_counts` is depending on the `buckets` you split for the histogram, you should initialize `cumulative_counts` according to the `buckets` and `data`

Reply:
Updated